batchspawner/batchspawner.py

```diff
@@ -353,6 +358,26 @@ def stop(self, now=False):
                              self.job_id, self.current_ip, self.port)
                 )
 
+    @gen.coroutine
```

Comment:
This should look like:

```python
@async_generator
async def progress(self):
    while True:
        ...
        await yield_({...})
```

This can't be a tornado coroutine because those are themselves generators, so they can't tell the difference between a yield that should be treated as an async yield item vs a yield that's actually a breakpoint.